ambry-mysql/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java

```diff
@@ -173,11 +173,11 @@ public synchronized Connection getDatabaseConnection(boolean needWritable) throw
       logger.debug("Still connected to {}", connectedEndpoint.getUrl());
     } else {
       // New connection established
+      closeActiveConnection();
       connectedEndpoint = endpointConnectionPair.getFirst();
+      activeConnection = endpointConnectionPair.getSecond();
       String qualifier = connectedEndpoint.isWriteable() ? "writable" : "read-only";
       logger.info("Connected to {} enpoint: {}", qualifier, connectedEndpoint.getUrl());
-      closeActiveConnection();
-      activeConnection = endpointConnectionPair.getSecond();
```

Comment:
What was the reason for the reordering?

Reply:
Since I am resetting 'connectedEndpoint' to null inside method closeActiveConnection(), I had to move closeActiveConnection() before assignment of 'connectedEndpoint' in line 176.